src/Tasks/Microsoft.NET.Build.Tasks/ResolveFrameworkReferences.cs

```diff
@@ -55,167 +51,173 @@ public class ResolveFrameworkReferences : TaskBase
         {
-            var knownFrameworkReferences = KnownFrameworkReferences.Select(item => new KnownFrameworkReference(item))
+            var knownFrameworkReferencesForTargetFramework = KnownFrameworkReferences.Select(item => new KnownFrameworkReference(item))
                 .Where(kfr => kfr.TargetFramework.Framework.Equals(TargetFrameworkIdentifier, StringComparison.OrdinalIgnoreCase) &&
                               NormalizeVersion(kfr.TargetFramework.Version) == NormalizeVersion(new Version(TargetFrameworkVersion)))
-                .ToDictionary(kfr => kfr.Name);
+                .ToList();
+
+            //  TODO: How to handle duplicate FrameworkReferences
+            var frameworkReferenceDict = FrameworkReferences.ToDictionary(fr => fr.ItemSpec);
 
             List<ITaskItem> packagesToDownload = new List<ITaskItem>();
             List<ITaskItem> runtimeFrameworks = new List<ITaskItem>();
             List<ITaskItem> targetingPacks = new List<ITaskItem>();
             List<ITaskItem> runtimePacks = new List<ITaskItem>();
-            List<string> unresolvedFrameworkReferences = new List<string>();
+            List<ITaskItem> unavailableRuntimePacks = new List<ITaskItem>();
 
-            string appHostPackPattern = null;
-            string appHostPackVersion = null;
-            string appHostRuntimeIdentifiers = null;
-
-            foreach (var frameworkReference in FrameworkReferences)
+            foreach (var knownFrameworkReference in  knownFrameworkReferencesForTargetFramework)
             {
-                KnownFrameworkReference knownFrameworkReference;
-                if (knownFrameworkReferences.TryGetValue(frameworkReference.ItemSpec, out knownFrameworkReference))
+                frameworkReferenceDict.TryGetValue(knownFrameworkReference.Name, out ITaskItem frameworkReference);
+
+                //  Get the path of the targeting pack in the targeting pack root (e.g. dotnet/ref)
+                TaskItem targetingPack = new TaskItem(knownFrameworkReference.Name);
+                targetingPack.SetMetadata(MetadataKeys.PackageName, knownFrameworkReference.TargetingPackName);
+
+                string targetingPackVersion = null;
+                if (frameworkReference != null)
+                {
+                    //  Allow targeting pack version to be overridden via metadata on FrameworkReference
+                    targetingPackVersion = frameworkReference.GetMetadata("TargetingPackVersion");
+                }
+                if (string.IsNullOrEmpty(targetingPackVersion))
                 {
-                    if (!string.IsNullOrEmpty(knownFrameworkReference.AppHostPackNamePattern))
+                    targetingPackVersion = knownFrameworkReference.TargetingPackVersion;
+                }
+                targetingPack.SetMetadata(MetadataKeys.PackageVersion, targetingPackVersion);
+
+                targetingPack.SetMetadata(MetadataKeys.RelativePath, "");
```

Comment:
Why do we need an explicit empty metadata value? Is this so that it can become "data" folder later, and also not the same place for all targeting packs all at once?

Reply:
I think it was for parity with the runtime and apphost packs.  However, the asset resolution has changed since then so I can remove it.